homeassistant/components/homekit/__init__.py

```diff
@@ -71,7 +71,7 @@ def handle_homekit_service_start(service):
     return True
 
 
-def get_accessory(hass, state, aid, config):
+def get_accessory(state, aid, config):
```

Comment:
Once all Accesory signature are the same you can simplify this:

```python
def get_accessory(hass, state, aid, config):
    """Take state and return an accessory object if supported."""
    if not aid:
        _LOGGER.warning('The entitiy "%s" is not supported, since it '
                        'generates an invalid aid, please change it.',
                        state.entity_id)
        return None

    a_type = None
    a_kwargs = {'aid'} = aid

    if state.domain == 'sensor':
        unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
        if unit in (TEMP_CELSIUS, TEMP_FAHRENHEIT):
            a_type = 'TemperatureSensor'
        elif unit == '%':
            a_type = 'HumiditySensor'

    elif state.domain == 'cover':
        # Only add covers that support set_cover_position
        features = state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
        if features & SUPPORT_SET_POSITION:
            a_type = 'WindowCovering'

    elif state.domain == 'alarm_control_panel':
        a_type = 'SecuritySystem'
        a_kwargs['alarm_code'] = config.get(ATTR_CODE)

    elif state.domain == 'climate':
        features = state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
        support_temp_range = SUPPORT_TARGET_TEMPERATURE_LOW | \
            SUPPORT_TARGET_TEMPERATURE_HIGH
        # Check if climate device supports auto mode
        a_kwargs['support_auto'] = bool(features & support_temp_range)
        a_type = 'Thermostat'

    elif state.domain == 'light':
        a_type = 'Light'

    elif state.domain in ('switch', 'remote', 'input_boolean', 'script'):
        a_type = 'Switch'
    
    if a_type is None:
        return None

    _LOGGER.debug('Add "%s" as "%s"', state.entity_id, state.name)
    return TYPES[a_type](hass, state.name, state.entity_id, **a_kwargs)

```